src/ig/IGExporter.ts

```diff
@@ -1338,96 +1342,169 @@ export class IGExporter {
+      }
+    });
+
+    // Add new copyrightLabel property
+    if (this.config.copyrightLabel) {
+      this.ig.copyrightLabel = this.config.copyrightLabel;
+    }
 
-      // Add new copyrightLabel property
-      if (this.config.copyrightLabel) {
-        this.ig.copyrightLabel = this.config.copyrightLabel;
+    // Add new versionAlgorithm property
+    if (this.config.versionAlgorithmString) {
+      this.ig.versionAlgorithmString = this.config.versionAlgorithmString;
+    } else if (this.config.versionAlgorithmCoding) {
+      this.ig.versionAlgorithmCoding = this.config.versionAlgorithmCoding;
+    }
+
+    // Add new dependsOn.reason property
+    this.ig.dependsOn?.forEach(dependency => {
+      const configDependency = this.config.dependencies?.find(
+        d => d.packageId === dependency.packageId
+      );
+      if (configDependency.reason) {
+        dependency.reason = configDependency.reason;
       }
+    });
+  }
 
-      // Add new versionAlgorithm property
-      if (this.config.versionAlgorithmString) {
-        this.ig.versionAlgorithmString = this.config.versionAlgorithmString;
-      } else if (this.config.versionAlgorithmCoding) {
-        this.ig.versionAlgorithmCoding = this.config.versionAlgorithmCoding;
+  // If an R4 IG uses any new properties from R5, they should be included as extensions.
+  // Converting between FHIR versions is documented here: http://hl7.org/fhir/2022Sep/versions.html#extensions
+  addR5ExtensionsToR4() {
+    // Any new properties added to the IG resource in R5 should be included on extensions in R4 IGs
+    this.ig.definition.resource.forEach(resource => {
+      // Assign IG.definition.resource.profile to an extensions if provided
+      const configEntry = this.config.resources?.find(
+        r => r.reference?.reference === resource.reference?.reference
+      );
+      if (configEntry?.profile != null) {
+        resource.extension = [
+          ...(resource.extension ?? []),
+          {
+            url: 'http://hl7.org/fhir/5.0/StructureDefinition/extension-ImplementationGuide.definition.resource.profile',
+            valueCanonical: configEntry.profile
+          }
+        ];
       }
+    });
 
-      // Add new dependsOn.reason property
-      this.ig.dependsOn?.forEach(dependency => {
-        const configDependency = this.config.dependencies?.find(
-          d => d.packageId === dependency.packageId
-        );
-        if (configDependency.reason) {
-          dependency.reason = configDependency.reason;
+    // Add new IG.definition.page.source[x] property to an extension if it is provided. No need to set a default like R5 needs to.
+    this.ig.definition.page.page.forEach(page => {
+      // this.ig.definition.page is the toc.html page we create
+      // All configured pages are at the next level, so start at that level
+      this.addPageSourceExtensionForR4(page, this.config.pages);
+    });
```

Comment:
Should we also do something with `definition.page.name`? I thought about if we should use it to set the R4 `nameUrl` property, but I think it's technically a little different. (In fact, I think maybe `sourceUrl` is closer to `nameUrl` than `name` it).  So... maybe if there is a `page.name` we should add an extension for it?

Reply:
Yeah, I think we should do something to handle `definition.page.name`. I think this is one of the places I missed converting some of the exiting R4 properties into the correct format. Instead of an extension, I think I'm leaning toward setting `page.nameUrl` equal to the value of `page.name`. When going from R4 to R5, we set the `page.name` to whatever is the `page.nameUrl` [here](https://github.com/FHIR/sushi/blob/c3f6501f590435cf43fa93af910e3be923cb1fde/src/ig/IGExporter.ts#L1510-L1520). But if `sourceUrl` is closer to `nameUrl` than `name`, maybe the R5 implementation is wrong. What do you think?

Reply:
Let's talk and try to work it out together. I agree it is confusing.